core/src/main/scala/kafka/utils/IteratorTemplate.scala

```diff
@@ -27,7 +29,7 @@ object FAILED extends State
  * Transliteration of the iterator template in google collections. To implement an iterator
  * override makeNext and call allDone() when there is no more items
  */
-abstract class IteratorTemplate[T] extends Iterator[T] with java.util.Iterator[T] {
+abstract class IteratorTemplate[T] extends Iterator[T] with ClosableIterator[T] {
```

Comment:
Understood. I think it's worth doing what you suggest. The reason is that we can then rely on static analysis to find any `AutoCloseable` instances where `close` is not called (either via `try/resources` or explicitly).